app/models/course/assessment/question.rb

```diff
@@ -38,4 +40,15 @@ def attempt(submission)
     return actable.attempt(submission) if actable && actable.self_respond_to?(:attempt)
     fail NotImplementedError, 'Questions must implement the #attempt method for submissions.'
   end
+
+  # Checks if the given submission has a correct answer for the question.
+  #
+  # @param [Course::Assessment::Submission] submission The submission which the answer should
+  #   belong to.
+  # @return [Boolean] True if the question has been finished in the submission.
+  def finished_by?(submission)
+    return actable.finished_by?(submission) if actable && actable.self_respond_to?(:finished_by?)
```

Comment:
@lowjoel Different question types should implement their own `finished_by`, `submission.finished?(question)` does not work for multiple question types.